### PR TITLE
Implement reCAPTCHA when joining sections

### PR DESCRIPTION
**Pull request by drew-beckmen**
Details in [Jira ticket](https://codedotorg.atlassian.net/si/jira.issueviews:issue-html/LP-1378/LP-1378.html)

## Links
- [jira](https://codedotorg.atlassian.net/si/jira.issueviews:issue-html/LP-1378/LP-1378.html)

VYrEcoQx0S

demo

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Comment by jmkulwik** [comment]
What was the decision around adding this on the student dashboard as well?

**Comment by drew-beckmen** [comment]
I think we should discuss with those who commented on the Jira to get their
input. It seems logical to implement it on student dashboard as well, but I
didn’t want to go into that without confirmation. I think it would be best
as a follow up change if it is needed.

File: dashboard/app/controllers/followers_controller.rb [hunk 1]
```diff
@@ -26,10 +25,14 @@ def student_register
     end
 
     Retryable.retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
-      if @user.save && @section&.add_student(@user)
-        sign_in(:user, @user)
-        redirect_to root_path, notice: I18n.t('follower.registered', section_name: @section.name)
-        return
+      if !verify_recaptcha
```
Comment (jamescodeorg) [inline dashboard/app/controllers/followers_controller.rb]: I'm not very familiar with this code, but it seems a little odd to me that the captcha verification would be inside the retryable block?  Maybe move this outside the retryable block?

**Comment by jmkulwik** [comment]
@drew-beckmen - are you still waiting on a sign-off for this one?

**Pull request closed without merging**